contrib/inventory/azure_rm.py

```diff
@@ -311,10 +313,17 @@ def __init__(self, args):
             tenant = self.credentials.get('tenant')
             if not tenant:
                 tenant = 'common'
-            self.azure_credentials = UserPassCredentials(self.credentials['ad_user'],
-                                                         self.credentials['password'],
-                                                         tenant=tenant,
-                                                         cloud_environment=self._cloud_environment)
+            # Use legacy client_id
+            LEGACY_CLIENT_ID = '04b07795-8ddb-461a-bbee-02f9e1bf7b46'
+
+            LOGIN_ENDPOINT = self._cloud_environment.endpoints.active_directory
+            RESOURCE = self._cloud_environment.endpoints.active_directory_resource_id
+            context = adal.AuthenticationContext(LOGIN_ENDPOINT + '/' + tenant)
+            self.azure_credentials = AdalAuthentication(context.acquire_token_with_username_password,
+                                                        RESOURCE,
+                                                        self.credentials['ad_user'],
+                                                        self.credentials['password'],
+                                                        LEGACY_CLIENT_ID)
```

Comment:
What's the legacy_client_id, seems it is no need: https://docs.microsoft.com/en-us/python/azure/python-sdk-azure-authenticate?view=azure-python#mgmt-auth-token

Reply:
tbh, I din't find any good reference for that constant but that seems to be the CLI Client ID _backed_ in old Azure CLI [azure-xplat-cli/lib/util/constants.js#L47](https://github.com/Azure/azure-xplat-cli/blob/436834d678dee1facd2e3b1d7b2f5af930847b90/lib/util/constants.js#L47)

The example in that page unfortunately is only using **acquire_token_with_client_credentials** which is for Service Principal.

Reply:
Please follow the docs.microsoft.com to use the Authentication

Reply:
@yujunz which documentation? like I told you, I followed the doc  / SDK. As I said in the PR (on top) this was shared by a guy from Microsoft in their SDK repo.

@lmazuel Can you check if this follows your recommendations?

Reply:
UserPasswordCredentials is supposed to be deprecated in favor of ServicePrincipalCredentials. However, we still provide it for backward compatibility.
The UserPassword flow requires a client_id. But usually people didn't have one (they just have their user/password). A very long time ago (I wasn't even at Microsoft yet), implementation was made to use by default the CLI 1.0 client_id, for people who are still using this flow. [This is what is done in current implementation in msrestazure as well](https://github.com/Azure/msrestazure-for-python/blob/ac8495ab2881e0dde9b65f165baa5dc6b5ec7e5b/msrestazure/azure_active_directory.py#L356-L357).

I would suggest you allow people to give their own client_id if using user/password, but to fallback to the one from CLI 1.0 if not provided (likely most common scenario)